hyperspy/external/matplotlib/collections.py

```diff
@@ -0,0 +1,271 @@
+    heights : array-like
+        The lengths of second axes.
+    angles : array-like
+        The angles of the first axes, degrees CCW from the x-axis.
+    units : {'points', 'inches', 'dots', 'width', 'height', 'x', 'y', 'xy'}
+        The units in which majors and minors are given; 'width' and
+        'height' refer to the dimensions of the axes, while 'x' and 'y'
+        refer to the *offsets* data units. 'xy' differs from all others in
+        that the angle as plotted varies with the aspect ratio, and equals
+        the specified angle only when the aspect ratio is unity.  Hence
+        it behaves the same as the `~.patches.Ellipse` with
+        ``axes.transData`` as its transform.
+    **kwargs
+        Forwarded to `Collection`.
+    """
+
+    _path_generator = mpath.Path.unit_circle
+
+
+class RectangleCollection(_CollectionWithWidthHeightAngle):
+    """
+    A collection of rectangles, drawn using splines.
+
+    Parameters
+    ----------
+    widths : array-like
+        The lengths of the first axes (e.g., major axis lengths).
+    heights : array-like
+        The lengths of second axes.
+    angles : array-like
+        The angles of the first axes, degrees CCW from the x-axis.
+    units : {'points', 'inches', 'dots', 'width', 'height', 'x', 'y', 'xy'}
+        The units in which majors and minors are given; 'width' and
+        'height' refer to the dimensions of the axes, while 'x' and 'y'
+        refer to the *offsets* data units. 'xy' differs from all others in
+        that the angle as plotted varies with the aspect ratio, and equals
+        the specified angle only when the aspect ratio is unity.  Hence
+        it behaves the same as the `~.patches.Ellipse` with
+        ``axes.transData`` as its transform.
+    **kwargs
+        Forwarded to `Collection`.
+
+    """
+    _path_generator = mpath.Path.unit_rectangle
+
+
+class TextCollection(_CollectionWithSizes):
+
+    def __init__(self, texts, sizes=None, **kwargs):
+        """
+        Parameters
+        ----------
+        texts : array-like
+            The texts of the collection.
+        angles : array-like
+            The angles of the first axes, degrees CCW from the x-axis.
+        **kwargs
+            Forwarded to `Collection`.
+        """
+        super().__init__(**kwargs)
+        self.set_sizes(sizes)
+        self._set_texts(texts)
+        self.set_transform(transforms.IdentityTransform())
+
+    def _set_texts(self, texts):
+        self._texts = texts
+        self._generate_path_from_text()
+
+    def set_texts(self, texts):
+        self._set_texts(texts)
+        self.stale = True
+
+    def get_texts(self, texts):
+        return self._texts
+
+    def _generate_path_from_text(self):
+        # For each TextPath, the position is at (0, 0) because the position
+        # will be given by the offsets values
+        self._paths = [TextPath((0, 0), text) for text in self._texts]
```

Comment:
@ericpre Just a note that current if you have a string all of the letters/numbers overlap because they all are set at the same position.  You can just set the same offset for each symbol but you might also have to do some things with kerning to make the spacing correct.

Reply:
Never mind ignore that... I just set it up wrong and it didn't fail because strings are iterable.